packages/webapp/pages/onboarding.tsx

```diff
@@ -229,7 +240,11 @@ export function OnboardPage(): ReactElement {
     }
 
     if (user?.infoConfirmed && activeScreen === OnboardingStep.Intro) {
-      router.replace(getPathnameWithQuery(webappUrl, window.location.search));
+      const afterAuth = params.get('afterAuth');
+      params.delete('afterAuth');
+      router.replace(
+        getPathnameWithQuery(afterAuth || webappUrl, params.toString()),
```

Comment:
```suggestion
        getPathnameWithQuery(afterAuth || webappUrl, params),
```

it accepts `URLSearchParams` as well